lib/spark.js

```diff
@@ -91,12 +89,21 @@ export default class Spark {
     console.log('Submitting retrieval...')
     const res = await this.#fetch(`https://spark.fly.dev/retrievals/${id}`, {
       method: 'PATCH',
-      body: JSON.stringify(stats),
+      body: JSON.stringify({
+        ...stats,
+        walletAddress: Zinnia.walletAddress
+      }),
       headers: {
         'Content-Type': 'application/json'
       }
     })
-    assertEquals(res.status, 200)
+    if (res.status !== 200) {
+      let body
+      try {
+        body = await res.text()
+      } catch {}
+      throw new Error(`Failed to submit retrieval: ${body}`)
```

Comment:
Maybe you can include the status code in the error message to better handle the case when the server does not provide any details in the response body? Maybe it does not matter, IDK.